app/controllers/placements/application_controller.rb

```diff
@@ -4,9 +4,24 @@ class Placements::ApplicationController < ApplicationController
 
   private
 
+  def set_school
+    @school = if current_user.support_user?
+                Placements::School.find(params.require(:school_id))
+              else
+                current_user.schools.find(params.require(:school_id))
+              end
+  end
+
+  def set_provider
+    @provider = if current_user.support_user?
+                  Placements::Provider.find(params.require(:provider_id))
+                else
+                  current_user.providers.find(params.require(:provider_id))
+                end
+  end
```

Comment:
This seems like an ideal use case for policy scopes. Although it looks like we might not actually have suitable scopes defined for these use cases yet.

Maybe one to follow up with in a separate PR to avoid doing too much here?

Reply:
I will address in https://github.com/DFE-Digital/itt-mentor-services/pull/1101